test/e2e/nstemplatetier_test.go

```diff
@@ -92,47 +92,148 @@ func TestNSTemplateTiers(t *testing.T) {
 	}
 }
 
-func TestUpdateToOtherRevisionOfTierTemplates(t *testing.T) {
-	// given
-	tierList := &toolchainv1alpha1.NSTemplateTierList{}
-	ctx, awaitility := testsupport.WaitForDeployments(t, tierList)
+func TestUpdateNSTemplateTier(t *testing.T) {
+	// in this test, we have 2 groups of users, configured with their own tier (both using the "basic" tier templates)
+	// then, the first tier is updated with the "advanced" templates, whereas the second one is updated using the "team" templates
+	// finally, all user namespaces are verified.
+	// So, in this test, we verify that namespace resources and cluster resources are updated, on 2 groups of users with different tiers ;)
+
+	count := 2*MaxPoolSize + 1
+	ctx, awaitility := testsupport.WaitForDeployments(t, &toolchainv1alpha1.NSTemplateTier{})
 	defer ctx.Cleanup()
 
-	// Create and approve "testing-template-updates" user
-	murName := "testingtemplateupdates"
-	userSignup := createAndApproveSignup(t, awaitility, murName)
+	// first group of users: the "cheesecake lovers"
+	cheesecakeTier, cheesecakeSyncIndexes := setupAccounts(t, ctx, awaitility, "cheesecake", "cheesecakelover%02d", count)
+	// second group of users: the "cookie lovers"
+	cookieTier, cookieSyncIndexes := setupAccounts(t, ctx, awaitility, "cookie", "cookielover%02d", count)
 
-	tierTemplatesByTier := getTierTemplatesMappedByTier(t, awaitility)
-	templatesFromAdvanced := createBasicTierTemplate(t, ctx, awaitility, tierTemplatesByTier["advanced"], "123abcd")
-	templatesFromTeam := createBasicTierTemplate(t, ctx, awaitility, tierTemplatesByTier["team"], "456efgh")
-	templatesFromBasic := createBasicTierTemplate(t, ctx, awaitility, tierTemplatesByTier["basic"], "789ijkl")
+	// when updating the "cheesecakeTier" tier with the "advanced" template refs (ie, same number of namespaces)
+	updateTemplateTier(t, awaitility, cheesecakeTier, "advanced")
+	// when updating the "cookie" tier with the "team" template refs (ie, different number of namespaces)
+	updateTemplateTier(t, awaitility, cookieTier, "team")
 
-	// wait for the user to be provisioned for the first time
-	verifyResourcesProvisionedForSignup(t, awaitility, userSignup, "basic")
+	// then
+	verifyResourceUpdates(t, awaitility, cheesecakeSyncIndexes, cheesecakeTier.Name, "advanced")
+	verifyResourceUpdates(t, awaitility, cookieSyncIndexes, cookieTier.Name, "team")
+}
```

Comment:
Do you think that we could also test updating the cookie tier back to basic to verify that the additional namespace will be created?
```suggestion
	// and when updating the "cookie" tier back to the "basic" template refs (ie, different number of namespaces)
	updateTemplateTier(t, awaitility, cookieTier, "basic")
	
	// then
	verifyResourceUpdates(t, awaitility, cookieSyncIndexes, cookieTier.Name, "basic")
}
```